mp4parse/src/boxes.rs

```diff
@@ -19,7 +20,40 @@ macro_rules! box_database {
                 }
             }
         }
+
+        impl Into<u32> for BoxType {
+            fn into(self) -> u32 {
+                use self::BoxType::*;
+                match self {
+                    $($boxenum => $boxtype),*,
+                    UnknownBox(t) => t,
+                }
+            }
+        }
+
+        impl fmt::Debug for BoxType {
+            fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+                let box_num: u32 = Into::into(self.clone());
+                write!(f, "{}", convert_to_fourcc(box_num))
+            }
+        }
+    }
+}
+
+pub fn convert_to_fourcc(number: u32) -> String {
+    let mut box_chars = Vec::new(); 
+    for x in 0..4 {
+        let c = (number >> x * 8 & 0x000000FF) as u8;
+        box_chars.push(c);
     }
+    box_chars.reverse();
+
+    let box_string = match String::from_utf8(box_chars) {
+        Ok(t) => t,
+        _ => String::from("error to get 4CC"),
+    };
+
+    box_string
```

Comment:
This is fine, but I think it would be better to create a new `Fourcc` type and use that instead of Strings everywhere. You can implement `From<BoxType>` and `From<u32>` to make conversion easy, and we'll have a little more type safety.

We could then write `brands.push(Fourcc::from(be_u32(src)?));` which is a little idiomatic than `convert_to_fourcc()`